src/modules/Channel/components/MessageFeedbackModal/hooks/useKeyDownV2.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useLayoutEffect } from 'react';
+
+type KeyDownCallbackMap = Record<string, (event: React.KeyboardEvent<HTMLDivElement>) => void>;
+
+interface UseKeyDownV2Props {
+  ref: React.RefObject<HTMLDivElement>;
+  keyDownCallbackMap: KeyDownCallbackMap;
+}
+
+export function useKeyDownV2(props: UseKeyDownV2Props): React.KeyboardEventHandler<HTMLDivElement> {
+  const { ref, keyDownCallbackMap } = props;
+
+  useLayoutEffect(() => {
+    ref.current?.focus();
+  }, [ref.current]);
+
+  const onKeyDown: React.KeyboardEventHandler<HTMLDivElement> = useCallback((event) => {
```

Comment:
```suggestion
  const onKeyDown: React.KeyboardEventHandler<HTMLDivElement> = usePreservedCallback((event) => {
```

Reply:
@bang9 keyDownCallbackMap must be a dependency so I think we have to use useCallback here

Reply:
This is the issue. using `usePreservedCallback` instead of `useCallback` will resolve this issue.

```tsx
useKeydown({
  ref,
  keyDownCallbackMap: {    // --> Inline object ==> Always new object
    Escape: () => { ... }
  }
})

const useKeydown = ({ keyDownCallbackMap }) => {
  const onKeyDown = useCallback = useCallback(() => {
    // ...
  },[keyDownCallbackMap]) // --> Always new object
  
  return onKeyDown // --> Always new callback
}
```

Reply:
oh it seems to work. That's very interesting that usePreservedCallback does not need dependency but useCallback does